### PR TITLE
Remove unnecessary uses of autograd.numpy.

### PR DESCRIPTION
**Pull request by smite**
**Context:**

The current version of PennyLane depends on Autograd, and uses it as its primary automatic differentiation (AD) framework. Autograd is not being maintained any more, and we are looking into supporting other AD frameworks on an equal basis.

Several modules in PL use `autograd.numpy`, a wrapped version of numpy with extra autograd functionality added, instead of the normal `numpy`. In most cases there seems to be no reason for this. As far as I can tell `autograd.numpy` functions are only required (instead of normal `numpy` functions) inside classical computational nodes that are meant to be differentiated using Autograd tools.

**Description of the Change:**

* Replaces `autograd.numpy` with plain `numpy` wherever possible.
* Adds a test for a case where `autograd.numpy` *is* required.

**Benefits:**

Starts the process of removing Autograd as a required dependency.

### CODE REVIEW

File: pennylane/_device.py [hunk 1]
```diff
@@ -90,7 +90,8 @@
 # pylint: disable=too-many-format-args
 import abc
 
-import autograd.numpy as np
+import numpy as np
+
```
Comment (josh146) [inline pennylane/_device.py]: lol why were we even using it here

File: tests/test_quantum_gradients.py [hunk 2]
```diff
@@ -17,7 +17,8 @@
 
 import pytest
 import autograd
-import autograd.numpy as np
+import autograd.numpy as anp  # only to be used inside classical computational nodes
+import numpy as np
```
Comment (josh146) [inline tests/test_quantum_gradients.py]: Nice :+1:

File: tests/test_quantum_gradients.py [hunk 3]
```diff
@@ -468,6 +468,44 @@ def d_error(p, grad_method):
         assert grad_fd1 == pytest.approx(grad_auto, abs=tol)
         assert grad_angle == pytest.approx(grad_auto, abs=tol)
 
+
+    def test_hybrid_gradients_numpy(self, qubit_device_2_wires, tol):
```
Comment (josh146) [inline tests/test_quantum_gradients.py]: Should we add `autograd` to the test name just to make it clear?

Comment (smite) [reply tests/test_quantum_gradients.py]: Sure.

File: pennylane/_device.py (hunk 1 above)
Comment (smite) [reply pennylane/_device.py]: Better safe than sorry? :)

**Pull request merged**